### PR TITLE
Fix website build errors both for Grafana and content mounted in Grafana Cloud

### PR DESCRIPTION
**Pull request by jdbaldry**
[Fix reference to Grafana CLI](https://github.com/grafana/grafana/commit/58edeeaba61383e077ecd11f5b730b2cd7374463) 

- Make relref for relative permalink
- Use relative path for unambiguous resolution
  
[Fix alerting relref anchor format](https://github.com/grafana/grafana/commit/8f1ca7cf8fab19fc768feef791d11d62d01cc0b2) 
  
[Avoid ambiguous relref lookups by forcing relative resolution](https://github.com/grafana/grafana/commit/a3a77b432e7c5b1b414c942cd85f9b68b55880e0) 
  
[Remove reference to non-existent shared page](https://github.com/grafana/grafana/commit/89af5b60f28a0f62fc6753548761f09e1e529e8a) 
  
[Fix links broken in Grafana Cloud using absolute relrefs](https://github.com/grafana/grafana/commit/f811b98c1c48d30664245db40cd5cf663b541b8a) 

By resolving the relref absolutely, it refers to the same location regardless of mounted directory. 
This means mounted content will link back to Grafana docs but I believe this is preferable to 404s.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

#### Prerequisites

Checkout each of the following as a subdirectory of your working directory:
- grafana/grafana branch `jdb/2023-03-fix-website-build-errors`
- grafana/cloud-docs branch `jdb/2023-03-fix-doc-validator`
- grafana/oncall ref `v0.0.39`
- grafana/incident branch `main`
- grafana/mimir branch `release-2.7`
- grafana/technical-documentation branch `main`
- grafana/website branch `jdb/2023-03-zero-docs-build-errors`

1. In the technical-documentation repository, run the `make-docs` script to build all documentation together:

   ```console
   $  ./scripts/make-docs \
        oncall \
        incident \
        mimir:v2.7.x \
        grafana:next \
        grafana:v9.4 \
        grafana-cloud \
        "arbitrary:$(pwd)/../website/scripts/docs/versions.js:/hugo/scripts/docs/versions.js" \
        "arbitrary:$(pwd)/../website/layouts/shortcodes/docs:/hugo/layouts/shortcodes/docs" \
        "arbitrary:$(pwd)/../website/content/docs/grafana/v8.1/packages_api:/hugo/content/docs/grafana/v8.1/packages_api"
   ```

1. Verify that Hugo logs no `WARN` or `ERROR` messages.

### CODE REVIEW

File: docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ Configure alerting behavior in the absence of data using information in the foll
...
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
 | Alerting                | Set alert rule state to `Alerting`. From Grafana 8.5, the alert rule waits for the entire duration for which the condition is true before firing. |
 | OK                      | Set alert rule state to `Normal`                                                                                                                  |
-| Error                   | Create a new alert `DatasourceError` with the name and UID of the alert rule, and UID of the datasource that returned no data as labels.          |
+| re                      | Error                                                                                                                                             | Create a new alert `DatasourceError` with the name and UID of the alert rule, and UID of the datasource that returned no data as labels. |
```
Comment (chri2547) [inline docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md:93]: This looks odd to me.

File: docs/sources/alerting/fundamentals/alert-rules/alert-rule-types.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Grafana supports several alert rule types, the following sections will explain t
 
 Grafana-managed rules are the most flexible alert rule type. They allow you to create alerts that can act on data from any of your existing data sources.
 
-In additional to supporting any datasource you can also add additional [expressions]({{< relref "../../../panels-visualizations/query-transform-data/expression-queries/" >}}) to transform your data and express alert conditions.
+In additional to supporting any datasource you can also add additional [expressions]({{< relref "/docs/grafana/latest/panels-visualizations/query-transform-data/expression-queries" >}}) to transform your data and express alert conditions.
```
Comment (chri2547) [inline docs/sources/alerting/fundamentals/alert-rules/alert-rule-types.md:19]:
```suggestion
In addition to supporting any data source, you can add additional [expressions]({{< relref "/docs/grafana/latest/panels-visualizations/query-transform-data/expression-queries" >}}) to transform your data and express alert conditions.
```

**Review by chri2547 — APPROVED** [review]
one question and a suggestion unrelated to fixing links (typos, couldn't help myself)

**Review by Eve832 — COMMENTED** [review]
Forgive me if I'm misremembering, but I thought that there was a fundamental problem with using paths that begin with `/` in relrefs?

**Review by Eve832 — APPROVED** [review]

File: docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md (hunk 1 above)
Comment (jdbaldry) [reply docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md:93]: Great catch! This is a multiple cursors error :D I'll fix it.

**Comment by jdbaldry** [comment]
> Forgive me if I'm misremembering, but I thought that there was a fundamental problem with using paths that begin with `/` in relrefs?

You're not misrembering, we did believe this for a while but the reasoning we had was mistaken.

Absolute paths as input to the `relref` parameter is definitely supported and works the same as specifying an absolute path as a partial URI in plain Markdown except we get the benefit of compile time link checking.

**Comment by Eve832** [comment]
one more question (I thought I left this comment already, but can't find it now). some relrefs have the trailing `/` removed and some don't

**Comment by jdbaldry** [comment]
> one more question (I thought I left this comment already, but can't find it now). some relrefs have the trailing `/` removed and some don't

Ideally, all relrefs should not have a trailing slash (another thing I've learnt this week) but having one isn't always a problem. In this case, all the relrefs still work as shown by the absence of any errors in https://drone.grafana.net/grafana/grafana/106149/1/7. 

I am going to make doc-validator prescribe that the relrefs don't have trailing slashes and then we can use the tool to identify and fix the occurences.

**Pull request merged**